flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/module/hive/HiveModuleDescriptorValidator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.module.hive;
+
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.descriptors.ModuleDescriptorValidator;
+
+/**
+ * Validator for {@link HiveModuleDescriptor}.
+ */
+public class HiveModuleDescriptorValidator extends ModuleDescriptorValidator {
+	public static final String MODULE_TYPE_HIVE = "hive";
+	public static final String MODULE_HIVE_VERSION = "hive-version";
+
+	@Override
+	public void validate(DescriptorProperties properties) {
+		super.validate(properties);
+		properties.validateValue(MODULE_TYPE, MODULE_TYPE_HIVE, false);
+		properties.validateString(MODULE_HIVE_VERSION, true, 1);
```

Comment:
should we verify that one of the hive versions that we have shims for?

Reply:
it is verified by hive shim